substrate/client/rpc/src/state/mod.rs

```diff
@@ -59,6 +59,15 @@ where
 		call_data: Bytes,
 	) -> Result<Bytes, Error>;
 
+	/// Call the runtime method at the block's parent state.
+	/// and pass the block to be replayed, followed by the call data.
+	fn debug_block(
```

Comment:
I wonder whether these could be pallet specific RPCs for pallet-revive instead of "state_debug_block" because from my understanding it supposed to be used for "ReviveApi_trace_tx" and ""ReviveApi_trace_block"?

Reply:
I thought it was pretty generic and potentially useful for other pallets that could setup tracing features in a similar way.

It can't just be a runtime API as it need to load the runtime and the blocks before passing it over to the runtime api that will handle this data

Reply:
In general I'm quite reluctant to add new stuff to the legacy RPCs but this is fine by me.

You didn't look into implementing this on-top of https://paritytech.github.io/json-rpc-interface-spec/api/archive_unstable_body.html?